thirdeye-server/src/main/java/ai/startree/thirdeye/resources/NamespaceConfigurationResource.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2024 StarTree Inc
+ *
+ * Licensed under the StarTree Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at http://www.startree.ai/legal/startree-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT * WARRANTIES OF ANY KIND,
+ * either express or implied.
+ * See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package ai.startree.thirdeye.resources;
+
+import static ai.startree.thirdeye.ResourceUtils.respondOk;
+
+import ai.startree.thirdeye.auth.ThirdEyeServerPrincipal;
+import ai.startree.thirdeye.service.NamespaceConfigurationService;
+import ai.startree.thirdeye.spi.Constants;
+import ai.startree.thirdeye.spi.api.NamespaceConfigurationApi;
+import ai.startree.thirdeye.spi.datalayer.dto.NamespaceConfigurationDTO;
+import io.dropwizard.auth.Auth;
+import io.micrometer.core.annotation.Timed;
+import io.swagger.v3.oas.annotations.OpenAPIDefinition;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.enums.SecuritySchemeIn;
+import io.swagger.v3.oas.annotations.enums.SecuritySchemeType;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.security.SecurityScheme;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+
+@Tag(name = "Namespace Configuration")
+@SecurityRequirement(name="oauth")
+@SecurityRequirement(name = Constants.NAMESPACE_SECURITY)
+@OpenAPIDefinition(security = {
+    @SecurityRequirement(name = "oauth"),
+    @SecurityRequirement(name = Constants.NAMESPACE_SECURITY)
+})
+@SecurityScheme(name = "oauth", type = SecuritySchemeType.APIKEY, in = SecuritySchemeIn.HEADER, paramName = HttpHeaders.AUTHORIZATION)
+@SecurityScheme(name = Constants.NAMESPACE_SECURITY, type = SecuritySchemeType.APIKEY, in = SecuritySchemeIn.HEADER, paramName = Constants.NAMESPACE_HTTP_HEADER)
+@Singleton
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class NamespaceConfigurationResource extends CrudResource<NamespaceConfigurationApi,
```

Comment:
Maybe it's ok to not extend CrudResource in this case.

1. We only need a GET and a PUT.
GET should always return a single element so enforcing it at the API level will force to limit potential implementation errors (eg namespace filter not working as expected)
For internal use we may want to perform stats on configuration across multiple namespaces but it's a very limited use case (not sure we will ever do this)

2. the problem when we override the CRUD ressources endpoints is the endpoints are still visible in swagger API. This makes the API harder to understand.

3. this will make this class way shorter

Reply:
Makes sense. simplifying both files

Reply:
Done